### ISSUE TITLE
Equivalent to prepareQuery() for updates

### ISSUE THREAD

**Issue by RichardWallis** [RDFLib/rdflib#1402]
Is there, or should there be, an equivalent function to the sparql `prepareQuery()` for use with updates?

Needing to run the same update lots of times, for performance reasons I ideally would like to prepare it only once.

Looking for a `prepareUpdate()` function I could not find one.   Browsing the source led me to this compromise:
```
        if re.search('INSERT', self.text, re.IGNORECASE) or re.search('DELETE', self.text, re.IGNORECASE):
            self.update = True
            self.compiled = translateUpdate(parseUpdate(self.text))
        else:
            self.update = False
            self.compiled = prepareQuery(self.text)
```
Which seems to work OK, but I would be more comfortable with an 'official' utility function.

**Comment by gjhiggins** [comment]
Closing as duplicate of #272, migrated to TO-DO discussion https://github.com/RDFLib/rdflib/discussions/1581
